sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -68,10 +68,10 @@ package object dsl {
     def unary_! : Predicate = Not(expr)
     def unary_~ : Expression = BitwiseNot(expr)
 
-    def + (other: Expression): Expression = Add(expr, other)
-    def - (other: Expression): Expression = Subtract(expr, other)
-    def * (other: Expression): Expression = Multiply(expr, other)
-    def / (other: Expression): Expression = Divide(expr, other)
+    def + (other: Expression): Expression = UnresolvedAdd(expr, other)
```

Comment:
do we have to change the DSL? mostly `+` means `Add`, we can write `UnresolvedAdd` directly when necessary.

Reply:
yea, I will change this back, I'am suffering tests fail with this change.